### ISSUE TITLE
Some checkboxes are coming as boolean value but there is 1 checkbox coming as checkbox thus breaking everything

### ISSUE THREAD

**Issue by FurkanGozukara** [gradio-app/gradio#7656]
This bug is driving me crazy

The checkbox is coming to the method as input as checkbox not as boolean and it enters inside if statement instead of the value is being False

How can I directly check value of checkbox?

![image](https://github.com/gradio-app/gradio/assets/19240467/c80a3f5a-9442-455e-878e-34b16baf387c)

Microsoft Windows [Version 10.0.19045.4123]
(c) Microsoft Corporation. All rights reserved.

R:\SUPIR_v26\SUPIR\venv\Scripts>activate

(venv) R:\SUPIR_v26\SUPIR\venv\Scripts>pip freeze
accelerate==0.27.2
aiofiles==23.2.1
aiohttp==3.9.3
aiosignal==1.3.1
altair==5.2.0
annotated-types==0.6.0
antlr4-python3-runtime==4.9.3
anyio==4.3.0
appdirs==1.4.4
async-timeout==4.0.3
attrs==23.2.0
bitsandbytes==0.43.0
braceexpand==0.1.7
certifi==2024.2.2
charset-normalizer==3.3.2
click==8.1.7
colorama==0.4.6
contourpy==1.2.0
cycler==0.12.1
diffusers==0.26.3
docker-pycreds==0.4.0
einops==0.7.0
einops-exts==0.0.4
exceptiongroup==1.2.0
facexlib==0.3.0
fastapi==0.110.0
ffmpy==0.3.2
filelock==3.13.1
filterpy==1.4.5
fonttools==4.49.0
frozenlist==1.4.1
fsspec==2024.2.0
ftfy==6.1.3
gitdb==4.0.11
GitPython==3.1.42
gradio==4.21.0
gradio_client==0.12.0
gradio_imageslider==0.0.18
h11==0.14.0
httpcore==1.0.4
httpx==0.27.0
huggingface-hub==0.21.4
idna==3.6
importlib_metadata==7.0.2
importlib_resources==6.1.3
Jinja2==3.1.3
joblib==1.3.2
jsonschema==4.21.1
jsonschema-specifications==2023.12.1
kiwisolver==1.4.5
kornia==0.7.1
lightning-utilities==0.10.1
llvmlite==0.42.0
Markdown==3.5.2
markdown-it-py==3.0.0
MarkupSafe==2.1.5
matplotlib==3.8.3
mdurl==0.1.2
mpmath==1.3.0
multidict==6.0.5
networkx==3.2.1
ninja==1.11.1.1
numba==0.59.0
numpy==1.26.4
omegaconf==2.3.0
open-clip-torch==2.24.0
openai-clip==1.0.1
opencv-python==4.9.0.80
orjson==3.9.15
packaging==23.2
pandas==2.2.1
pillow==10.2.0
protobuf==4.25.3
psutil==5.9.8
pydantic==2.6.3
pydantic_core==2.16.3
pydub==0.25.1
Pygments==2.17.2
pyparsing==3.1.2
python-dateutil==2.9.0.post0
python-multipart==0.0.9
pytorch-lightning==2.2.1
pytz==2024.1
PyYAML==6.0.1
referencing==0.33.0
regex==2023.12.25
requests==2.31.0
rich==13.7.1
rpds-py==0.18.0
ruff==0.3.2
safetensors==0.4.2
scikit-learn==1.4.1.post1
scipy==1.12.0
semantic-version==2.10.0
sentencepiece==0.2.0
sentry-sdk==1.41.0
setproctitle==1.3.3
shellingham==1.5.4
six==1.16.0
smmap==5.0.1
sniffio==1.3.1
starlette==0.36.3
sympy==1.12
threadpoolctl==3.3.0
timm==0.9.16
tokenizers==0.15.2
tomlkit==0.12.0
toolz==0.12.1
torch==2.2.0+cu121
torchaudio==2.2.0+cu121
torchmetrics==1.3.1
torchvision==0.17.0+cu121
tqdm==4.66.2
transformers==4.38.2
triton @ https://huggingface.co/MonsterMMORPG/SECourses/resolve/main/triton-2.1.0-cp310-cp310-win_amd64.whl
typer==0.9.0
typing_extensions==4.10.0
tzdata==2024.1
urllib3==2.2.1
uvicorn==0.28.0
wandb==0.16.4
wcwidth==0.2.13
webdataset==0.2.86
websockets==11.0.3
xformers==0.0.24
yarl==1.9.4
zipp==3.17.0

(venv) R:\SUPIR_v26\SUPIR\venv\Scripts>

### Severity

Blocking usage of gradio

**Comment by FurkanGozukara** [comment]
it was parameter passing error :/
